src/utils/deployFreeFlow.ts

```diff
@@ -0,0 +1,87 @@
+import type {default as VM, Env as Env} from "../types/vm";
+import createNetwork from "./createNetwork";
+import type {IProfile} from "../types/Profile";
+import deploy from "./deploy";
+import type {IStore} from "../stores/currentDeployment";
+import checkVMExist from "./prepareDeployment";
+import {Network} from "../types/kubernetes";
+import type FreeFlow from "../types/freeflow";
+
+const {MachineModel, MachinesModel, GatewayNameModel} = window.configs?.grid3_client ?? {};
+
+export default async function deployFreeFlow(data: FreeFlow, profile: IProfile, type: IStore["type"]) {
+    const {envs, disks, ...base} = data;
+    const {vmName, threeBotUserId, flist, cpu, memory, entrypoint, network, rootFs} = base;
+    const {publicIp, planetary, nodeId} = base;
+
+    const vm = new MachineModel();
+    vm.name = vmName;
+    vm.node_id = nodeId;
+    vm.disks = disks;
+    vm.memory = memory;
+    vm.rootfs_size = rootFs;
+    vm.flist = flist;
+    vm.entrypoint = entrypoint;
+    vm.cpu = cpu;
+    vm.planetary = planetary;
+    vm.public_ip = publicIp;
+
+
+    vm.env = createEnvs(envs);
+
+    const vms = new MachinesModel();
+    vms.name = threeBotUserId;
```

Comment:
the 3bot name is still used as the deployment name. i got this error, is it supposed to be only one instance for each user? 

![Untitled](https://user-images.githubusercontent.com/32527347/196473684-93d3ee8d-3742-4f93-ba09-6d607739ea29.png)
note: `omarabdulaziz` is my 3bot name not the name i entered in the deployment name field

Reply:
Oops, forgot to change that name too ^^ 

Yeah, noticed this problem too, even if I deleted all of my contracts on devnet, a new NamedGateway with a previous created name (which is already) deleted, cant be deployed.

See: https://github.com/threefoldtech/test_feedback/issues/299